packages/fcl-wc/src/service.js

```diff
@@ -0,0 +1,170 @@
+import QRCodeModal from "@walletconnect/qrcode-modal"
+
+export const makeServicePlugin = client => ({
+  name: "fcl-service-walletconnect",
+  type: "discovery-service",
+  services: makeWcServices(client),
+})
+
+const makeServiceStrategy = client => {
+  return ({service, body, opts}) => {
+    return new Promise(async (resolve, reject) => {
+      if (typeof client === "undefined") {
+        throw new Error("WalletConnect is not initialized")
+      }
+      let session
+      const onResponse = resp => {
+        try {
+          if (typeof resp !== "object") return
+
+          switch (resp.status) {
+            case "APPROVED":
+              resolve(resp.data)
+              break
+
+            case "DECLINED":
+              reject(`Declined: ${resp.reason || "No reason supplied"}`)
+              break
+
+            case "REDIRECT":
+              resolve(resp)
+              break
+
+            default:
+              reject(`Declined: No reason supplied`)
+              break
+          }
+        } catch (error) {
+          console.error("execWcRPC onResponse error", error)
+          throw error
+        }
+      }
+
+      const onClose = () => {
+        reject(`Declined: Externally Halted`)
+      }
+
+      if (client.session.length) {
+        const lastKeyIndex = client.session.keys.length - 1
+        session = client.session.get(client.session.keys[lastKeyIndex])
+      }
+      if (session == null) {
+        let pairing = {topic: service?.provider?.address ?? undefined}
```

Comment:
I think this will return undefined with the optional chaining if no match

Reply:
It needs to be undefined if no match. Using null [here](https://github.com/onflow/fcl-js/blob/gsantos/wc-service-plugin/packages/fcl-wc/src/service.js#L100) will fail

Reply:
so here's the rub. Ultimately need a (better) way to differentiate between a generic WC service, and an existing pairing

Reply:
Changed `provider.address` to “WalletConnect” for generic

Reply:
If generic and paired wallet service use same address, i believe Discovery dedupes right @chasefleming 
Left pairing.topic for now on pair service but added provider.uid to actually handle the topic.